project/project.go

```diff
@@ -21,27 +21,27 @@ import (
 	"github.com/rs/zerolog/log"
 )
 
-// RelPath returns the dir relative to project's root.
-func RelPath(root, dir string) string {
+// PrjAbsPath converts the file system absolute path absdir into an absolute
+// project path.
+func PrjAbsPath(root, absdir string) string {
 	log.Trace().
-		Str("action", "RelPath()").
-		Str("dir", dir).
+		Str("action", "PrjAbsPath()").
+		Str("dir", absdir).
 		Str("root", root).
 		Msg("Trim path to get relative dir.")
 
-	d := strings.TrimPrefix(dir, root)
-
+	d := strings.TrimPrefix(absdir, root)
```

Comment:
Wouldn't it be safer to use [filepath.Rel](https://pkg.go.dev/path/filepath#Rel) and then just prepend a filepath.Separator ? Right now if absdir is not relative to root it would just return absdir itself (nothing trimmed). Maybe this sort of invariant is guaranteed somewhere else.

Reply:
Not sure if [filepath.Rel](https://cs.opensource.google/go/go/+/refs/tags/go1.17.6:src/path/filepath/path.go;l=266) is what we want here, as it deals with lots of file system details like Windows volumes... and it also fails but in our case (if there's no bug) the `root` is always a substring of `absdir`. If we make this fail for a programming error, it'll clumsy the code and give strange error messages (user seeing an error that's not his fault). Maybe we can panic here if `absdir` is not really a superstring of `root` ?

Reply:
If we guarantee that on another level it is also OK to not check again, like it is the caller responsability to ensure that.